### PR TITLE
feat: add a bundle target for web browsers

### PR DESCRIPTION
**Pull request by jlipps**
This allows us to run e.g. `npm run build:prod:browser`, and get output in `dist-browser` which can be served as a set of static files from a web server, enabling Appium Inspector to run in a browser rather than only as a standalone Electron app.

To accomplish this I had to implement some of the node-only or electron-only interfaces using web technologies, but otherwise it was fairly straightforward.

### CODE REVIEW

File: app/renderer/polyfills/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+let log, settings, clipboard, shell, remote, ipcRenderer,
+    i18NextBackend, i18NextBackendOptions;
+
+function buildForBrowser () {
+  if (process.env.BUILD_BROWSER) {
+    return true;
+  }
+
+  if (typeof navigator !== 'undefined') {
```
Comment (mykola-mokhnach) [inline app/renderer/polyfills/index.js:9]: this could be simplified to `return typeof navigator !== 'undefined' && !/electron/i.test(navigator.userAgent)`

**Review by mykola-mokhnach — APPROVED** [review]

**Comment by mykola-mokhnach** [comment]
shall we do something about tests?

**Review by KazuCocoa — COMMENTED** [review]
Great idea, yey
This change itself seems good to me, but CI is failing. I guess something electron's `app`...?

File: app/renderer/polyfills/index.js (hunk 1 above)
Comment (jlipps) [reply app/renderer/polyfills/index.js:9]: ah yes good point. i was debugging something and left the debugging structure.

File: app/renderer/polyfills/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+let log, settings, clipboard, shell, remote, ipcRenderer,
+    i18NextBackend, i18NextBackendOptions;
+
+function buildForBrowser () {
+  if (process.env.BUILD_BROWSER) {
```
Comment (mykola-mokhnach) [inline app/renderer/polyfills/index.js:5]: we could reduce this whole function body to a single return

**Review by mykola-mokhnach — APPROVED** [review]

File: app/renderer/polyfills/index.js (hunk 2 above)
Comment (jlipps) [reply app/renderer/polyfills/index.js:5]: i know, i found this a bit easier to follow

**Pull request merged**